.circleci/config.yml

```diff
@@ -17,17 +24,17 @@ jobs:
             # when lock files change, use increasingly general
             # patterns to restore cache
             - &cache_key # yamllint disable-line rule:line-length
-              python-3.8-packages-v1-{{ .Branch }}-{{ checksum "requirements.in" }}-{{ checksum "requirements.txt" }}
+              python-<< pipeline.parameters.python-version >>-packages-v1-{{ .Branch }}-{{ checksum "requirements.in" }}-{{ checksum "requirements.txt" }}
               # yamllint disable-line rule:line-length
-            - python-3.8-packages-v1-{{ .Branch }}-{{ checksum "requirements.in" }}-
-            - python-3.8-packages-v1-{{ .Branch }}-
-            - python-3.8-packages-v1-master-
-            - python-3.8-packages-v1-
+            - python-<< pipeline.parameters.python-version >>-packages-v1-{{ .Branch }}-{{ checksum "requirements.in" }}-
+            - python-<< pipeline.parameters.python-version >>-packages-v1-{{ .Branch }}-
+            - python-<< pipeline.parameters.python-version >>-packages-v1-master-
+            - python-<< pipeline.parameters.python-version >>-packages-v1-
       - &build
         run:
           name: Build
           command: |
-            python3.8 -m venv venv/
+            python3 -m venv venv/
```

Comment:
Do we even need to specify python3 binary here? Or is using python instead enough?